src/components/DealProposalForm.tsx

```diff
@@ -89,6 +91,33 @@ const FormInput = ({
   const startBlockRealTime = blockToTime(startBlock, currentBlock, currentBlockTimestamp);
   const endBlockRealTime = blockToTime(endBlock, currentBlock, currentBlockTimestamp);
 
+  const [marketBalance, setMarketBalance] = useState<string>("");
+
+  const { collatorWsApi: api } = useCtx();
+
+  // This useEffect fetches market balance for the selected account so it can be displayed when it is selected.
+  useEffect(() => {
+    const fetchMarketBalance = async () => {
+      if (selectedAccount && api) {
+        try {
+          setMarketBalance("(loading...)");
+
+          const result = await api.query.market.balanceTable(selectedAccount.address);
+          const json = result.toJSON() as Record<string, unknown>;
+          const free = (json.free as string) ?? "0";
+          setMarketBalance(free);
+        } catch (err) {
+          console.error("Error fetching market balance:", err);
+          setMarketBalance("Error");
```

Comment:
This is super sus. A better approach would be to use the same we're using for the connection status: use an union type to model the error/non-error state